### DISCUSSION TITLE
AsDiscreted Extremely Slow

### DISCUSSION THREAD

**Discussion by sunshichen** [Project-MONAI/MONAILabel#653] [Q&A] [answered]
I tried to test monailabel follow the official guidline on a remote server and use 3Dslicer as annotation client. How ever it seems unusual that AsDiscreted in post-processing is extremely slow, the log as below:

![1645766526(1)](https://user-images.githubusercontent.com/23454839/155658705-90d0f661-0754-47e0-bf7c-739c10e2435b.png)

I used deepedit_multilabel sample on Ubuntu18.04 with Nvidia GTX 1080Ti.

**Comment by SachidanandAlle** [comment, marked as answer]
You are running this transform across multiple labels (512, 512, 164) is a large image.
Now, in the latest release candidates, the post transform are run on GPU to make it faster.
https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/infers/deepedit.py#L112
`EnsureTyped(keys="pred", device=data.get("device") if data else None)`

If still a bottleneck, please raise an issue in monai repo for this transform
